### ISSUE TITLE
[Feature Request] Release conversion script to allow custom model conversion

### ISSUE THREAD

**Issue by baysonfox** [0seba/VoxCPMANE#5]
Looking to run VoxCPM1.5 with custom LoRA adapters (a finetuned voice, no architecture changes) on Apple Silicon. 
Since the repository currently provides only the pre-compiled .mlmodelc files, I cannot use my own LoRA or merge my lora into the base model and then convert into mlpackage -> compile to mlmodelc

It would be great to be able to convert models by ourselves, and TY for the amazing work for porting to ANE!

**Comment by 0seba** [comment]
Hi sorry for the delay I was away from my computer? Looking at the VoxCPM repo I see that the finetuning can add adapters to the LM, DiT and projection layers so I may be able to provide a script to convert those to CoreML. Do you happen to have the finetuned weights that you could share on huggingface so I can confirm which models need to be reconverted?

**Comment by baysonfox** [comment]
Sorry for delay on my side! Christmas, New Year and all sorts of things etc.
Here's the LoRA:
[https://huggingface.co/baysonfox/VoxCPM-Amiya-LoRA](https://huggingface.co/baysonfox/VoxCPM-Amiya-LoRA)

FYI, the default finetune script from upstream seems to apply LoRA to on only BaseLM and DiT.

```json
"lora_config": {
  "enable_lm": true,
  "enable_dit": true,
  "enable_proj": false,
  "r": 64,
  "alpha": 32,
  "dropout": 0.0,
  "target_modules_lm": [
    "q_proj",
    "v_proj",
    "k_proj",
    "o_proj"
  ],
  "target_modules_dit": [
    "q_proj",
    "v_proj",
    "k_proj",
    "o_proj"
  ],
  "target_proj_modules": [
    "enc_to_lm_proj",
    "lm_to_dit_proj",
    "res_to_dit_proj"
  ]
}
```

**Comment by yukiarimo** [comment]
+1

**Comment by 0seba** [comment]
thanks for the interest! i am sorry i have not been able to deliver, i am working on an improved conversion pipeline to support more models and for other tasks. no ETA for tts models =(

**Comment by yukiarimo** [comment]
@0seba Do you know how to train VoxCPM’s VAE? I tried with simple loss (L1) but it doesn’t work

**Comment by bzowskiNK** [comment]
Hi — we fine-tuned VoxCPM2 with stress LoRA (enable_lm + enable_dit, enable_proj=false, r=32) and fused it into a full openbmb/VoxCPM2-layout folder. We run VoxCPMANE2 on Apple Silicon for stream/bake and need the same CoreML conversion pipeline that produced seba/VoxCPM2ANE-Preview (multifunction BaseLM/ResidualLM lengths 1/8/16/32/64/128). Is qeml / the conversion scripts available, even privately or as a paid convert of our fused weights?

**Comment by 0seba** [comment]
@bzowskiNK hi that's so cool to hear! right now i don't have a clear decision of how to move forward with this pacakge so can you upload the lora weights, as .safetensors and ideally make it clear which LoRA corresponds to which weight, you could use separate safetensors file for the different modules i wouldn't mind, and i'll create the converted package for you.

since i have to quantize part of the BaseLM and the DiT in order to reach real time generation speed, i'll have to do some quantization of your weights as well
